test/data_file_test.cpp

```diff
@@ -103,12 +102,78 @@ TEST(DataFileTest, LoadTypicalFile)
 		{ "", "2", "3" },
 		{ "1", "2", "" },
 		{ "1", "", "3" },
-		{ "" } // file ends with a newline, parsing returns a single empty field
 	};
 
 	TestFileContents(dataFile, expectedFields);
 }
 
+TEST(DataFileTest, LoadFileWithNoTrailingNewline)
+{
+	auto result = LoadDataFile("txtdata\\lf_no_trail.tsv");
+	ASSERT_TRUE(result.has_value()) << "Unable to load lf_no_trail.tsv";
+
+	DataFile &dataFile = result.value();
+	EXPECT_EQ(dataFile.size(), 32) << "File size should be reported in code units";
+
+	std::vector<std::vector<std::string_view>> expectedFields {
+		{ "Test", "Empty", "Values" },
+		{ "", "2", "3" },
+		{ "1", "2", "" },
+		{ "1", "", "3" },
+	};
+
+	TestFileContents(dataFile, expectedFields, GetFieldResult::Status::EndOfRecord, GetFieldResult::Status::NoFinalTerminator);
+}
+
+std::string_view mapError(DataFile::Error error)
+{
+	switch (error) {
+	case DataFile::Error::NotFound:
+		return "not found";
+	case DataFile::Error::OpenFailed:
+		return "cannot open";
+	case DataFile::Error::BadRead:
+		return "cannot read contents";
+	default:
+		return "unexpected error";
+	}
+}
+
+TEST(DataFileTest, LoadEmptyFile)
+{
+	//GTEST_SKIP() << "Loading 0 length assets is not supported by AssetHandle, trying to load this file fails with Error::BadRead";

```

Comment:
I'm just guessing here:
```suggestion
	// GTEST_SKIP() << "Loading 0 length assets is not supported by AssetHandle, trying to load this file fails with Error::BadRead";
```

Reply:
Yeah this needs to be removed, I added a workaround that meant I no longer need to skip this test but forgot to delete the comment/line. Got sidetracked trying to fix the grammar.

Reply:
```suggestion
```